pkg/config/analysis/analyzers/authz/authorizationpolicies.go

```diff
@@ -57,10 +57,54 @@ func (a *AuthorizationPoliciesAnalyzer) Analyze(c analysis.Context) {
 	c.ForEach(collections.IstioSecurityV1Beta1Authorizationpolicies.Name(), func(r *resource.Instance) bool {
 		a.analyzeNoMatchingWorkloads(r, c, podLabelsMap)
 		a.analyzeNamespaceNotFound(r, c)
+		a.analyzeL7DenyAuthzPolicies(r, c)
 		return true
 	})
 }
 
+func (a *AuthorizationPoliciesAnalyzer) analyzeL7DenyAuthzPolicies(r *resource.Instance, c analysis.Context) {
+	ap := r.Message.(*v1beta1.AuthorizationPolicy)
+	apNs := r.Metadata.FullName.Namespace.String()
+
+	if ap.Action == v1beta1.AuthorizationPolicy_DENY {
+		for _, rule := range ap.Rules {
+			if rule.From != nil || rule.To != nil {
+				tcpRulesInFrom := false
+				tcpRulesInTo := false
+				for _, from := range rule.From {
+					if from.Source.GetPrincipals() != nil ||
+						from.Source.GetNotPrincipals() != nil || from.Source.GetIpBlocks() != nil ||
+						from.Source.GetNotIpBlocks() != nil || from.Source.GetNamespaces() != nil ||
+						from.Source.GetNotNamespaces() != nil || from.Source.GetRemoteIpBlocks() != nil ||
+						from.Source.GetNotRemoteIpBlocks() != nil {
+						tcpRulesInFrom = true
+					}
+				}
+
+				for _, to := range rule.To {
+					if to.Operation.GetPorts() != nil || to.Operation.GetNotPorts() != nil {
+						tcpRulesInTo = true
+					}
+				}
+
+				if !tcpRulesInFrom && !tcpRulesInTo {
+					if ap.Selector != nil && ap.Selector.MatchLabels != nil {
+						for key, val := range ap.Selector.MatchLabels {
+							label := key + ":" + val
+							m := msg.NewRejectedTCPTrafficAuthz(r, label, apNs)
+							c.Report(collections.IstioSecurityV1Beta1Authorizationpolicies.Name(), m)
```

Comment:
Isn't this what people are supposed to be doing? Apply a MatchLabel...? 

The issue was users were making them too broad I thought?

If we are warning for matchlabel, then what  is a user supposed to change?

Reply:
also this will report many times for each label, should just be joining all labels together

Reply:
Issue was not with the broader policies. If someone has a L7 deny policy on matchlabel `app:bookinfo`, then all the tcp traffic for that workload will be denied. From the 2nd comment, did you mean attach all the labels on one single policy together right ?

Reply:
So how exactly am I supposed to write an L7 deny policy then?

As a user, I get this warning - what do I do?

Reply:
add ports whenever you're adding any L7 deny rules.

Reply:
I am gonna add a scenario about this usecase in our docs.

Reply:
won't this warn even if they have port?

Btw, if it's just statically validating a single resource it should be in validation.go. This makes it run as part of the webhook and show warning with kubectl.

Reply:
it wont warn if there is a port, but do we want the warning with every kubectl command ? It's gonna be too much for a user, if they find something wrong with their TCP traffic they can check istioctl analyze. Let me know what you think about this.

Reply:
It's either worth warning about it it's not... if it's not useful and noisy we shouldn't have it in analyzer either.

My expectations was that it was only warning when there was no workloadSelector btw

Reply:
The issue doesn't have anything to do with workload selector, so it's better if we raise warning for all type of policies. Generally a user runs a istioctl analyze just once/twice but runs the kubectl command a lot more times. That's my thinking behind this, with kubectl we are going to do the publicity of this behavior whereas with istioctl analyze only the victims will get to know about this.

Reply:
IMO the sooner we can warn the better.

The relation for workloadSelector was really the same as port - if you are already scoping it finely, the problem is small/non-existant. The big issue is accidentally blocking all TCP in the namespace or entire cluster.

Note `port` is not safe either, really - a port can have TCP and HTTP on it. So this is all heuristics anyways

Reply:
Yeah I agree it is still little confusing with this. But the only way to resolve this is by adding ports (no matter they belong to http or tcp). Based on your comments we have 2 options - 

1) Add a warning in `isitoctl-analyze` only for the L7 deny rules applied on the namespace level. 
2) Add a warning with every kubectl command for both workload and ns level policies if L7 deny rule is there.

LMK which one should I go for ?

Reply:
sg, thanks